step-1.5/public/index.js

```diff
@@ -0,0 +1,29 @@
+(function() {
+
+  var redBtn = document.getElementById('red-button');
+  var blueBtn = document.getElementById('blue-button');
+  var yellowBtn = document.getElementById('yellow-button');
+  var body = document.querySelector('body');
+
+  function httpGetRequest(colour) {
+    var xhr = new XMLHttpRequest();
+    xhr.onreadystatechange = () => {
```

Comment:
Revert to ES5 `function () {}` instead of ES6 arrow function in front-end JS to avoid issues with browser compatibility. Also, plz make changes to the file in all the steps.